addon/components/frost-table.js

```diff
@@ -54,12 +81,34 @@ export default Component.extend({
     return columns.map((column, index) => Object.assign({index}, column))
   },
 
+  @readOnly
+  @computed()
+  _isSelectable () {
+    return !isNone(this.get('onSelectionChange'))
+  },
+
   // == Functions =============================================================
 
+  setShift (event) {
+    if (!this.isDestroyed) {
+      this.set('_isShiftDown', event.shiftKey)
+    }
+  },
+
   // == DOM Events ============================================================
 
   // == Lifecycle Hooks =======================================================
 
+  init () {
+    this._super(...arguments)
+    const itemKey = this.get('itemKey')
+    if (itemKey) {
+      this.set('_itemComparator', function (lhs, rhs) {
```

Comment:
- Selection isn't required for a table (or list) so an itemKey isn't required.  
- `id` is not a guaranteed property
- `_itemComparator` is bound to `itemKey` because we want to compare for equality using that key most of the time instead of attempting a strict comparison on the object itself